_studio/mfx_lib/encode_hw/hevc/linux/base/hevcehw_base_va_packer_lin.cpp

```diff
@@ -635,6 +637,20 @@ void VAPacker::InitAlloc(const FeatureBlocks& /*blocks*/, TPushIA Push)
 
             return false;
         });
+        cc.AddPerPicMiscData[VAEncMiscParameterTypeMaxSliceSize].Push([](
+            CallChains::TAddMiscData::TExt
+            , const StorageR& global
+            , const StorageR&
+            , std::list<std::vector<mfxU8>>& data)
+        {
+            const auto& par = Base::Glob::VideoParam::Get(global);
+            const mfxExtCodingOption2& CO2 = ExtBuffer::Get(par);
+            auto& maxSliceSize_param = AddVaMisc<VAEncMiscParameterMaxSliceSize>(VAEncMiscParameterTypeMaxSliceSize, data);
```

Comment:
I think we don't need to submit VAEncMiscParameterMaxSliceSize if CO2.MaxSliceSize == 0
What about reset, do we need to resubmit it on reset?

Reply:
I don't think we need to resubmit it on reset.

Reply:
Do we have such restriction in driver so we cannot reset CO2.MaxSliceSize to a new value? Seems such functionality is present in legacy code.

Reply:
Thanks @vilichev , we don't have such restriction in driver. So I upload a new commit to add it in reset.